common/src/main/java/us/myles/ViaVersion/protocols/protocol1_9to1_8/storage/EntityTracker.java

```diff
@@ -59,13 +59,13 @@ public EntityTracker(UserConnection user) {
     }
 
     public UUID getEntityUUID(int id) {
-        if (uuidMap.containsKey(id)) {
-            return uuidMap.get(id);
-        } else {
-            UUID uuid = UUID.randomUUID();
+        UUID uuid = uuidMap.get(id);
+        if (uuid == null) {
+            uuid = UUID.randomUUID();
             uuidMap.put(id, uuid);
```

Comment:
ComputeIfAbsent would probably be faster here.

Reply:
The project is set to Java 7 afaik

Reply:
Oh yeah just checked and it is, that's unfortunate ^^